airflow/config_templates/config.yml

```diff
@@ -375,6 +375,32 @@ core:
       type: string
       example: "path.to.CustomXCom"
       default: "airflow.models.xcom.BaseXCom"
+    xcom_objectstorage_path:
```

Comment:
I wonder if this should be more general. Having two configs that only make sense if `xcom_backend` is set to a specific backend seems awkward. I wonder if we can make the backend take arguments directly… One quick idea is to provide a backend _factory_ in Airflow instead, and tell the user to actually create the backend class dynamically…

```python
# $AIRFLOW_HOME/plugins/xcom.py

CustomBackend = create_objectstorage_backend(
    storage_path="s3://conn_id@bucket/path",
    threshold=1_000_000,
)
```

```ini
# airflow.cfg
xcom_backend = xcom.CustomBackend
```

Reply:
Sure, of course I had in mind to make it part of core and the default. Is there any that does the factory pattern? I don't think it should reside in plugins as it should be easy to reach / config.

Reply:
I like it the way it is at the moment - just enabling it by configuration is great. No need to inject custom code for a factory needed or a plugin.

Reply:
Yeah, I think the same. I'd like to keep it. There is no real harm and common.io is 'close' enough to core.

Reply:
The closest precedent we have is probably executors? We can create a new config section `[objectstorage_xcom]` that allows setting configs specific to the XCom backend. Although the section would be injected by a provider in the executor’s case, while we don’t have an equivalent here.

Reply:
But those config are already moved to the provider config :) .